src/etc/rc.newwanipv6

```diff
@@ -121,6 +122,26 @@ system_hosts_generate();
 
 $cacheip = @file_get_contents($cacheip_file);
 
+// If on DS-Lite, try to find a valid / usable AFTR address (either from DHCPv6 or manually configured)
+if (interface_dslite_is_enabled($interface)) {
+    // Try to read dhcp option 64 (AFTR name) from the environment
+    $aftr_addr = trim(getenv("raw_dhcp_option_64"));
+
+    // convert the raw argument, which is hex, to a string
+    if (!empty($aftr_addr)) {
+        $aftr_addr = pack("H*", $aftr_addr);
+        log_error("Got AFTR from DHCPv6 response: {$aftr_addr}");
+    }
+
+    if (!empty(trim($config['interfaces'][$interface]['dslite_aftr_addr']))) {
```

Comment:
if `dslite_aftr_addr` takes precedence over `getenv("raw_dhcp_option_64")` it's better to make it explicit in the call flow.  if dslite_aftr_addr then use this, else something like.

`$aftr_addr = !empty(trim(getenv("raw_dhcp_option_64"))) ? pack("H*",  trim(getenv("raw_dhcp_option_64"))) : null;
`

Reply:
I actually think it's pretty ugly to use the same code over and over again like the `getenv( .... )` but well, you asked for it. Wait for typos :D